File: SimpleDate.java
public class SimpleDate {
    private int day;
    private int month;
    private int year;

    public SimpleDate(int day, int month, int year) {
        this.day = day;
        this.month = month;
        this.year = year;
    }

    public int getDay() {
        return this.day;
    }

    public int getMonth() {
        return this.month;
    }

    public int getYear() {
        return this.year;
    }

    public boolean equals(Object compared) {
        if (this == compared) {
            return true;
        }

        if (!(compared instanceof SimpleDate)) {
            return false;
        }

        SimpleDate comparedSimpleDate = (SimpleDate) compared;

        if (this.day == comparedSimpleDate.day &&
            this.month == comparedSimpleDate.month &&
            this.year == comparedSimpleDate.year) {
                return true;
            }

        return false;
    }

    @Override
    public String toString() {
        return this.day + "." + this.month + "." + this.year;
    }

    public boolean before(SimpleDate compared) {
        if (this.year < compared.year) {
            return true;
        }

        if (this.year == compared.year &&
            this.month < compared.month) {
                return true;
        }

        if (this.year == compared.year &&
            this.month == compared.month &&
            this.day < compared.day) {
                return true;
            }

        return false;
    }


    public void advance() {
        this.day++;

        if (this.day > 30) {
            this.day = 1;
            this.month++;

            if (this.month > 12) {
                this.month = 1;
                this.year++;
            }
        }
    }

    public void advance(int howManyDays) {
        for (int i = 0; i < howManyDays; i++) {
            this.advance();
        }
    }

    public SimpleDate afterNumberOfDays(int days) {
        SimpleDate clonedDate = new SimpleDate(this.day, this.month, this.year);
        
        clonedDate.advance(days);
        return clonedDate;
    }
}

File: Practice/SimpleDate.cs
public class SimpleDate
{
    private int day;
    private int month;
    private int year;

    public SimpleDate(int day, int month, int year)
    {
        this.day = day;
        this.month = month;
        this.year = year;
    }

    public int GetDay()
    {
        return this.day;
    }

    public int GetMonth()
    {
        return this.month;
    }

    public int GetYear()
    {
        return this.year;
    }

    public override bool Equals(object? obj)
    {
        if (this == obj)
        {
            return true;
        }

        if (!(obj is SimpleDate))
        {
            return false;
        }

        SimpleDate comparedDate = (SimpleDate)obj;

        if (this.day == comparedDate.day &&
            this.month == comparedDate.month &&
            this.year == comparedDate.year)
        {
            return true;
        }

        return false;
    }

    public override string ToString()
    {
        return this.day + "." + this.month + "." + this.year;
    }
}